lib/src/Uncertainty/Model/EllipticalDistribution.cxx

```diff
@@ -467,26 +465,52 @@ void EllipticalDistribution::update()
   const UnsignedInteger dimension = getDimension();
   if (dimension > 1)
   {
-    // Build the shape matrix
+    // Compute the shape matrix
     shape_ = R_;
     for (UnsignedInteger i = 0; i < dimension; ++i)
       for (UnsignedInteger j = 0; j <= i; ++j)
         shape_(i, j) *= sigma_[i] * sigma_[j];
-    // Compute its Cholesky factor
-    cholesky_ = shape_.computeCholesky();
-
+    // Try to compute the Cholesky factor of the shape matrix
+    try
+    {
+      cholesky_ = shape_.computeCholesky();
+    }
+    // In the case where the matrix is not numerically SPD, try a unique regularization loop
+    // as it should succeed (otherwise the covariance matrix is truly non SPD)
+    catch(...)
+    {
+      Scalar largestEV = 0.0;
+      (void) shape_.getImplementation()->computeLargestEigenValueModuleSym(largestEV, 10, 1e-2);
+      for (UnsignedInteger i = 0; i < dimension; ++i)
+        R_(i, i) += largestEV * SpecFunc::Precision;
+      // This time throw if the decomposition fails
+      try
+        {
+          cholesky_ = R_.computeCholesky();
+        }
+      catch (...)
```

Comment:
looks like it could be the specific nonspdexception we have

Reply:
For each of the exceptions defined in Exception.cxx, I counted in how many files they are used:
```
493 InvalidArgumentException
125 NotYetImplementedException
 72 InternalException
 64 InvalidDimensionException
 59 NotDefinedException
 14 OutOfBoundException
  8 FileNotFoundException
  6 FileOpenException
  5 NotSymmetricDefinitePositiveException
  5 InvalidRangeException
  4 StudyFileParsingException
  3 ConfigurationFileParsingException
  3 XMLParserException
  2 XMLException
  2 DynamicLibraryException
  2 ObjectNotInStudyException
```
As you see, only 6 different types of exceptions are actually used. They were defined at a time where we planed to declare all the kind of exception each function/method should throw, but it was a nightmare. As a result, we still have all these exceptions even if most of them are marginal. So I am not sure if it is a good idea to revival some of them. If it adds an actual semantic to the code I am ready to use them, otherwise I will stick on the 6 first ones.

Reply:
grep -rl  "(\.\.\.)" lib/src | wc -l gives 27 files, and you can check that 25 of them are try{} catch (...) . Looks like we have some work here ;-)

Reply:
If I look at the code in MatrixImplementation::computeCholesky(), it is an InternalException. I propose to fix it as well as EllipticalDistribution.